py/picca/cf.py

```diff
@@ -558,12 +559,16 @@ def fill_t123(r1,r2,ang,w1,w2,z1,z2,c1d_1,c1d_2,w123,t123_loc,same_half_plate):
     for k in range(w.sum()):
         i1 = bins[k]%n1
         j1 = (bins[k]-i1)//n1
-        w123[ba[k]]+=we[k]
-        t123_loc[ba[k],ba[k]]+=we[k]/zw[k]
+
+        wAll[ba[k]] += we[k]
+        nb[ba[k]] += 1
+        T1[ba[k],ba[k]] += we[k]/zw[k]
+
         for l in range(k+1,w.sum()):
             i2 = bins[l]%n1
             j2 = (bins[l]-i2)//n1
             prod = c1d_1[i1,i2]*c1d_2[j1,j2]
-            t123_loc[ba[k],ba[l]]+=prod
-            t123_loc[ba[l],ba[k]]+=prod
+            T2[ba[k],ba[l]] += prod
+            T2[ba[l],ba[k]] += prod
 
```

Comment:
@londumas there's a problem here with the definition of `T1, T2, T3`. `T1` should contain only the product of the variances, `T2` a variance times a c1d and `T3` the product of two c1d. From what I see you have `T1` and `T3` but `T2` is missing.

Reply:
@ngbusca, fixed.